### ISSUE TITLE
A11y_Azure Data Studio Nov2023_Security_New Login Preview_Voiceover: Voiceover is announcing incorrect state as "Unticked" for selected/ticked checkbox in "Server Roles" table.

### ISSUE THREAD

**Issue by Mana-swi** [microsoft/azuredatastudio#25158]
**Please do not close this bug. This bug should only be closed by Trusted Tester after verification.**
[Check out Accessibility Insights!](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-stfe%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458850501301&sdata=mxhokIKNMb22llsjXHLgU3XZibj1Qfx37rpY4PU2sfE%3D&reserved=0) - Identify accessibility bugs before check-in and make bug fixing faster and easier.

### GitHub Tags:
#A11yTCS; #A11yMAS; #A11ySev2; #DesktopApp; #WCAG1.3.1; #Info and Relationships; #FTP; #SH_Azure Data Studio_Nov2023_Mac_Dec2023; #SQL Azure Data Studio; #MAC; #VoiceOver;

### Environment Details:
Application Name: Azure Data Studio
Operating System: macOS Sonoma 14.2.1
Azure data studio 1.47.0
VoiceOver

### Repro Steps:

1. Install Azure data studio application and open it
2. Make sure to enable "Preview" feature in settings.
3. Turn on voiceover.
4. Tab till "Server" and press enter to expand the tree.
5. Tab till "Security" and press enter.
6. Tab till "Login" and press enter.
7. Tab till any login and pressF10 to open context menu.
8. Tab till "New Login Preview" and press enter.
9. Tab till "Server Roles" table.
10. Tab till any Checkbox and press control-option-space to select it.
11. You will observe that Voiceover is announcing incorrect state as "Unticked" for selected/ticked checkbox in "Server Roles" table.

### Actual Result:
Voiceover is announcing incorrect state as "Unticked" for selected/ticked checkbox in "Server Roles" table.

Note: Similar issue is observed for all checkbox in table throughout the application.

### Expected Result:
Voice over should announce the correct state as "ticked" when any checkbox is selected/ticked. 

### User Impact:
Screen Reader users will get incorrect information about the status of checkbox as voiceover is announcing incorrect state.

### Attachment:
![Voiceover is announcing incorrect state as Unticked](https://github.com/microsoft/azuredatastudio/assets/155441466/6784da6f-c800-445a-8401-cc8b222eec31)

**Comment by kisantia** [comment]
dev notes: looks similar to https://github.com/microsoft/azuredatastudio/issues/22187, might be fixed by switching to using the declarative table component

**Comment by kisantia** [comment]
@Mana-swi this has been fixed. Please install the l[atest insiders build](https://github.com/microsoft/azuredatastudio?tab=readme-ov-file#try-out-the-latest-insiders-build-from-main-branch) and verify the fix.

**Comment by kisantia** [comment]
@Mana-swi this has been fixed. Please install the l[atest insiders build](https://github.com/microsoft/azuredatastudio?tab=readme-ov-file#try-out-the-latest-insiders-build-from-main-branch) and verify the fix.

**Comment by Mana-swi** [comment]
Hi @kisantia we are unable to download the latest Insiders build for Mac from provided link above. Can you please reshare it or help us with the build details. Thanks!

**Comment by kisantia** [comment]
@Mana-swi there was a brief outage. The download links should be working again now.

**Comment by Mana-swi** [comment]
#Regressed:02-05-24; Reopened;

Verified the issue in below environment 
Azure data Studio Version: 1.47.1 (Universal)
VS Code: 1.82.0
Electron: 25.8.4
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.2.0

The issue is still repro'ing, please find below attachment for reference. Thanks!

![25158_Voiceover is announcing incorrect state as Unticked for ticked checkbox](https://github.com/microsoft/azuredatastudio/assets/155441466/a3a01588-131d-4c5c-b1a4-a95f4543e613)

**Comment by Mana-swi** [comment]
Regressed:02-06-24;

Hi @corivera 
Reverified the issue in below environment
Version: 1.48.0-insider (Universal)
VS Code: 1.82.0
Electron: 25.9.8
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.2.0
MacOS: sonoma 14.2.1

The issue is only fixed for "New login Preview" dialog checkboxes, for rest all other dialogs where the checkboxes are present it's still repro'ing. Please find attachment for reference. Thanks!
![checkbox](https://github.com/microsoft/azuredatastudio/assets/155441466/d4e11673-701c-4130-ae2e-9703f4b6bf67)

**Comment by kisantia** [comment]
@smartguest reassigning this to you, since you're already looking at converting other tables to declarative

**Comment by kisantia** [comment]
@Mana-swi can you please try testing again with the [latest insiders build](https://github.com/microsoft/azuredatastudio?tab=readme-ov-file#try-out-the-latest-insiders-build-from-main-branch)? The other tables have also been fixed.

**Comment by smartguest** [comment]
Currently Fixed, although there may be additional unnecessary repeated information from the screen reader (which can be easily be addressed): 
![Server Roles NVDA Speech](https://github.com/microsoft/azuredatastudio/assets/18018452/1f5c71db-765a-401a-b761-68b5cdd6fd8a)

**Comment by Mana-swi** [comment]
#closed; #Regressed:03-13-24;

Verified the issue in below environment
Version: 1.49.0-insider (Universal)
Commit: abf8372bbe27427541a40f1f82308652fcac2e3c
Date: 2024-03-12T05:53:53.949Z
VS Code: 1.82.0
Electron: 25.9.8
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.3.0
MacOS: sonoma 14.3.1

The issue is fixed, hence closing the bug. Please find attachment for reference. Thanks!
![voiceover announcing correctly for ticked checkbox](https://github.com/microsoft/azuredatastudio/assets/155441466/42557f47-8ac1-4b08-baec-43ce4abf75b5)
